miio/viomivacuum.py

```diff
@@ -279,113 +344,393 @@ def mop_mode(self) -> ViomiMode:
+        # params: [edge_state, 0]
+        # - edge: see ViomiEdgeState
+        # - 2: pause cleaning
+        if not self._cache["edge_state"]:
+            self._cache["edge_state"] = self.get_properties(["mode"])
+        self.send("set_mode", self._cache["edge_state"] + [2])
+
+    @command()
+    def stop(self):
+        """FIXME: validate that Stop cleaning."""
+        # params: [edge_state, 0]
+        # - edge: see ViomiEdgeState
+        # - 0: stop cleaning
+        if not self._cache["edge_state"]:
+            self._cache["edge_state"] = self.get_properties(["mode"])
+        self.send("set_mode", self._cache["edge_state"] + [0])
+
+    @command(click.argument("mode", type=EnumType(ViomiMode)))
+    def clean_mode(self, mode):
+        """Set the cleaning mode.
+
+        [vacuum, vacuumAndMop, mop, cleanzone, cleanspot]
+        """
+        self.send("set_mop", [mode.value])
 
     @command(click.argument("speed", type=EnumType(ViomiVacuumSpeed)))
     def set_fan_speed(self, speed: ViomiVacuumSpeed):
-        """Set fanspeed [silent, standard, medium, turbo]."""
+        """Set fanspeed.
+
+        [silent, standard, medium, turbo]
+        """
         self.send("set_suction", [speed.value])
 
-    @command(click.argument("watergrade"))
+    @command(click.argument("watergrade", type=EnumType(ViomiWaterGrade)))
     def set_water_grade(self, watergrade: ViomiWaterGrade):
-        """Set water grade [low, medium, high]."""
+        """Set water grade.
+
+        [low, medium, high]
+        """
         self.send("set_suction", [watergrade.value])
 
-    @command()
-    def home(self):
-        """Return to home."""
-        self.send("set_charge", [1])
+    # The following commands are available on the SETTINGS page
+    # in the Mi Home Android app
 
-    @command(
-        click.argument("direction", type=EnumType(ViomiMovementDirection)),
-        click.option(
-            "--duration",
-            type=float,
-            default=0.5,
-            help="number of seconds to perform this movement",
-        ),
-    )
-    def move(self, direction, duration=0.5):
-        """Manual movement."""
-        start = time.time()
-        while time.time() - start < duration:
-            self.send("set_direction", [direction.value])
-            time.sleep(0.1)
-        self.send("set_direction", [ViomiMovementDirection.Stop.value])
+    # MISSING cleaning history
 
-    @command(click.argument("mode", type=EnumType(ViomiMode)))
-    def clean_mode(self, mode):
-        """Set the cleaning mode."""
-        self.send("set_mop", [mode.value])
+    @command()
+    def get_scheduled_cleanup(self):
+        """Not implemented yet."""
+        # Needs to reads and understand the return of:
+        # self.send("get_ordertime", [])
+        # [id, enabled, repeatdays, hour, minute, ?, ? , ?, ?, ?, ?, nb_of_rooms, room_id, room_name, room_id, room_name, ...]
+        return "Not Implemented yet."
```

Comment:
`raise NotImplementedError()` for not-yet implemented functionality.

Reply:
done